app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt

```diff
@@ -211,6 +212,28 @@ class InstalledAddonDetailsFragment : Fragment() {
         }
     }
 
+    private fun bindAllowInPrivateBrowsingSwitch(view: View) {
+        val switch = view.allow_in_private_browsing_switch
+        switch.isChecked = addon.isAllowedInPrivateBrowsing()
+        switch.setOnCheckedChangeListener { v, isChecked ->
+            val addonManager = v.context.components.addonManager
+            switch.isClickable = false
+            view.remove_add_on.isEnabled = false
+            addonManager.setAddonAllowedInPrivateBrowsing(
+                addon,
+                isChecked,
+                onSuccess = {
+                    runIfFragmentIsAttached {
+                        this.addon = it
+                        switch.isClickable = true
+                        switch.isChecked = isChecked
+                        view.settings.isVisible = shouldSettingsBeVisible()
+                        view.remove_add_on.isEnabled = true
```

Comment:
I'm not sure if we need these lines? Is there any specific reason? :)

Reply:
Ok, we are changing  ` view.remove_add_on.isEnabled = false` when the switch is clicked and checking it back in `onSuccess` that means the we need to handle `onError` because if I'm error happens 
` view.remove_add_on.isEnabled = false` will persisted

Reply:
We are getting new `addon` instance reference so these are safety UI "locks" similarly to enable/disable.